packages/core/src/config/options-marker.ts

```diff
@@ -0,0 +1,65 @@
+import type { JSONSchema7 } from 'json-schema';
+import { tokens } from 'typed-inject';
+import { StrykerOptions } from '@stryker-mutator/api/core';
+import { commonTokens } from '@stryker-mutator/api/plugin';
+import { Logger } from '@stryker-mutator/api/logging';
+import { PropertyPathBuilder, findUnserializables } from '@stryker-mutator/util';
+
+import { coreTokens } from '../di';
+import { isWarningEnabled } from '../utils/object-utils';
+
+markOptions.inject = tokens(commonTokens.options, coreTokens.validationSchema, commonTokens.logger);
+
+/**
+ * Performs additional validation on the Stryker options to mark unusual behavior with a warning.
+ * Namely when a value isn't serializable or when unknown options are passed in.
+ */
+export function markOptions(options: StrykerOptions, schema: JSONSchema7, log: Logger): StrykerOptions {
+  markUnknownOptions(options, schema, log);
+  markUnserializableOptions(options, log);
+  return options;
+}
+
+function markUnknownOptions(options: StrykerOptions, schema: JSONSchema7, log: Logger): void {
```

Comment:
is explicit return type needed here? We are not exporting that function

Reply:
Nope, it isn't. But it is allowed, isn't it? I'll remove these explicit return types.

Reply:
Explicit return types are bugprone. So if we don't have to, better let's not do it :D